src/server/qgsfcgiserverresponse.cpp

```diff
@@ -23,15 +23,141 @@
 #include <fcgi_stdio.h>
 #include <QDebug>
 
+#include <unistd.h>
+#include "qgslogger.h"
+
+#ifdef Q_OS_LINUX
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netinet/tcp.h>
+#endif
+
+#ifdef Q_OS_LINUX
 //
-// QgsFcgiServerResponse
+// QgsFCGXStreamData copied from libfcgi FCGX_Stream_Data
 //
+typedef struct QgsFCGXStreamData
```

Comment:
It's probably a naive question but why is it necessary to redefine the structure in this case?

Reply:
Did you try something like `FCGX_Request *r = static_cast<FCGX_Request*>(FCGI_stdin->fcgx_stream->data);`?

Reply:
no because `data` is a QgsFCGXStreamData (originally found [here](https://github.com/FastCGI-Archives/fcgi2/blob/874b148dfeadf8ebadff89cf228e63f836e509a3/libfcgi/fcgiapp.c#L1333)) disguised in a `void *`